### PR TITLE
Add mute icon

### LINKED ISSUE #345: Add mute button to routing mode
image

### PR DESCRIPTION
**Pull request by sarahlensing**
Adds a mute icon to the route view

Closes #345

### CODE REVIEW

File: app/src/main/kotlin/com/mapzen/erasermap/view/ButtonView.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.widget.ImageView
+import android.widget.RelativeLayout
+import com.mapzen.erasermap.R
+
+/**
+ * Generic class for buttons that need an icon centered on bg
+ */
+public abstract class ButtonView(context: Context, attrs: AttributeSet) : RelativeLayout(context, attrs) {
```
Comment (ecgreb) [inline app/src/main/kotlin/com/mapzen/erasermap/view/ButtonView.kt]: :+1:

File: app/src/main/kotlin/com/mapzen/erasermap/view/ButtonView.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    public abstract fun idForImage():Int
+
+    val background: ImageView by lazy { findViewById(R.id.background) as ImageView }
+    val image: ImageView by lazy { findViewById(idForImage()) as ImageView }
+
+    init {
+        (context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater)
+                .inflate(idForLayout(), this, true)
+    }
+}
```
Comment (ecgreb) [inline app/src/main/kotlin/com/mapzen/erasermap/view/ButtonView.kt]: GitHub gets angry if you don't put a newline at the end of files. We try to do that so we don't have to look at these formatting errors.

File: app/src/main/kotlin/com/mapzen/erasermap/view/MainActivity.kt [hunk 3]
```diff
@@ -13,10 +13,7 @@ import android.view.Menu
 import android.view.MenuItem
 import android.view.View
 import android.view.inputmethod.EditorInfo
-import android.widget.Button
-import android.widget.ImageButton
-import android.widget.RadioButton
-import android.widget.Toast
+import android.widget.*
```
Comment (ecgreb) [inline app/src/main/kotlin/com/mapzen/erasermap/view/MainActivity.kt]: Prefer single class imports over * imports.

You can configure this in Android Studio under File > Settings > Editor > Code Style > Kotlin > Imports.

File: app/src/main/kotlin/com/mapzen/erasermap/view/MainActivity.kt [hunk 4]
```diff
@@ -288,6 +299,22 @@ public class MainActivity : AppCompatActivity(), MainViewController, RouteCallba
...
+        //Update cached value of routePresenter.isMuted
+        val routePresenter = routeModeView.routePresenter
+        routePresenter?.onMuteClicked()
+
+        val muted = (routePresenter?.isMuted() == true)
+        muteView.setMuted(!muted)
+
+        //Actually mute or unmute the speakerbox based on current value
+        if (muted) {
+            routeModeView.voiceNavigationController!!.mute()
```
Comment (ecgreb) [inline app/src/main/kotlin/com/mapzen/erasermap/view/MainActivity.kt]: Prefer `?.` over `!!.` when referencing nullable variables. Kotlin makes it harder to get NPEs but if you really want one `!!.` is one of the best way to get them ;)

https://kotlinlang.org/docs/reference/null-safety.html

File: app/src/test/kotlin/com/mapzen/erasermap/presenter/MainPresenterTest.kt [hunk 5]
```diff
@@ -374,6 +374,13 @@ public class MainPresenterTest {
         assertThat(mainController.routeLine).isEqualTo(route)
     }
 
+    @Test fun onMuteClick_shouldToggleMute() {
+        val muted = mainController.muted
+        presenter.onMuteClick()
+        val mutedAfter = mainController.muted
+        assertThat(muted != mutedAfter)
```
Comment (ecgreb) [inline app/src/test/kotlin/com/mapzen/erasermap/presenter/MainPresenterTest.kt]: Fluent assertion syntax requires a actual, expected, and a condition predicate to work correctly.

```kotlin
assertThat(actual).condition(expected)
```

In this case you would want to replace what you have with:

```kotlin
assertThat(mutedAfter).isNotEqualTo(muted)
```

File: app/src/test/kotlin/com/mapzen/erasermap/presenter/RoutePresenterTest.kt [hunk 6]
```diff
@@ -39,6 +39,13 @@ public class RoutePresenterTest {
         assertThat(routeEngine.route).isEqualTo(route2)
     }
 
+    @Test fun onRouteStart_shouldResetMuted() {
+        routePresenter.setMuted(true)
+        val route = Route(getFixture("valhalla_route"))
+        routePresenter.onRouteStart(route)
+        assertThat(routePresenter.isMuted() == false)
```
Comment (ecgreb) [inline app/src/test/kotlin/com/mapzen/erasermap/presenter/RoutePresenterTest.kt]:
```kotlin
assertThat(routePresenter.isMuted()).isFalse()
```

File: app/src/main/kotlin/com/mapzen/erasermap/view/ButtonView.kt (hunk 2 above)
Comment (sarahlensing) [inline app/src/main/kotlin/com/mapzen/erasermap/view/ButtonView.kt]: Fixed.

File: app/src/main/kotlin/com/mapzen/erasermap/view/MainActivity.kt (hunk 3 above)
Comment (sarahlensing) [inline app/src/main/kotlin/com/mapzen/erasermap/view/MainActivity.kt]: Configured and fixed.

File: app/src/main/kotlin/com/mapzen/erasermap/view/MainActivity.kt (hunk 4 above)
Comment (sarahlensing) [inline app/src/main/kotlin/com/mapzen/erasermap/view/MainActivity.kt]: Fixed

File: app/src/test/kotlin/com/mapzen/erasermap/presenter/MainPresenterTest.kt (hunk 5 above)
Comment (sarahlensing) [inline app/src/test/kotlin/com/mapzen/erasermap/presenter/MainPresenterTest.kt]: Fixed

File: app/src/test/kotlin/com/mapzen/erasermap/presenter/RoutePresenterTest.kt (hunk 6 above)
Comment (sarahlensing) [inline app/src/test/kotlin/com/mapzen/erasermap/presenter/RoutePresenterTest.kt]: Fixed.

**Pull request merged**